app/page/send/index.tsx

```diff
@@ -1,50 +1,39 @@
 import * as ethers from "ethers"
-import { useCallback, useContext, useState, type ChangeEvent } from "react"
-import { Link } from "wouter"
+import { useState, type ChangeEvent } from "react"
+import { useRoute } from "wouter"
+import { navigate } from "wouter/use-hash-location"
 
 import { StepBackHeader } from "~app/component/stepBackHeader"
-import { ProviderContext } from "~app/context/provider"
+import { TokenItem } from "~app/component/tokenItem"
+import { TokenList } from "~app/component/tokenList"
 import { Path } from "~app/path"
-import { useAccount, useTokens } from "~app/storage"
-import { getChainName, getErc20Contract } from "~packages/network/util"
-import address from "~packages/util/address"
-import number from "~packages/util/number"
+import { useAccount } from "~app/storage"
+import { getUserTokens } from "~app/util/getUserTokens"
 import { type Token } from "~storage/local/state"
-import type { BigNumberish, HexString } from "~typing"
-
-const SelectToken = ({ tokens, onSelectToken }) => {
-  const handleAssetChange = async (event: ChangeEvent<HTMLSelectElement>) => {
-    const tokenAddress = event.target.value
-    console.log(`tokenAddress: ${tokenAddress}`)
-    const token = tokens.find((token) => token.address === tokenAddress)
-    onSelectToken(token)
-  }
+import type { BigNumberish, HexString, Nullable } from "~typing"
 
+const SelectToken = ({ setTokenSelected }) => {
+  const tokens = getUserTokens()
   return (
-    <div className="flex">
-      <label className="col-span-1">Asset:</label>
-      <select
-        id="asset"
-        value={tokens[0].address}
-        className="col-span-4 border w-full outline-none border-gray-300"
-        onChange={handleAssetChange}>
-        {tokens.map((token) => {
-          const balance = number.formatUnitsToFixed(
-            token.balance,
-            token.decimals
-          )
-          return (
-            <option key={token.address} value={token.address}>
-              {token.symbol}: {balance} {token.symbol}
-            </option>
-          )
-        })}
-      </select>
-    </div>
+    <>
+      <StepBackHeader title="Select Token" />
+      <TokenList className="pt-[16px]">
+        {tokens.map((token, index) => (
+          <TokenItem
+            key={index}
+            token={token}
+            onClick={() => {
+              setTokenSelected(token)
+              navigate(`/send/${token.address}`)
```

Comment:
Q: Must we `navigate` at the end? Wouldn't `setTokenSelected` trigger re-render to next step for us?

Reply:
[remove navigate when clicking TokenItem](https://github.com/pilagod/waallet-extension/pull/170/commits/a6a751bbf60dd3108c7d5e49d9ed2ae5f0dbe048)